### PR TITLE
ADEN-6015 Refactored TestAdsOnSpecialOasis

### PR DESCRIPTION
**Pull request by nandy-andy**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsContent.java#L72](https://github.com/Wikia/selenium-tests/blob/bb83574a5b7d809d716ff75620974f352aa8e2ec/src/test/java/com/wikia/webdriver/common/contentpatterns/AdsContent.java#L72): Define a constant instead of duplicating this literal "#BOTTOM_LEADERBOARD" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [AdsContent.java#L74](https://github.com/Wikia/selenium-tests/blob/bb83574a5b7d809d716ff75620974f352aa8e2ec/src/test/java/com/wikia/webdriver/common/contentpatterns/AdsContent.java#L74): Define a constant instead of duplicating this literal "#TOP_LEADERBOARD" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [AdsContent.java#L75](https://github.com/Wikia/selenium-tests/blob/bb83574a5b7d809d716ff75620974f352aa8e2ec/src/test/java/com/wikia/webdriver/common/contentpatterns/AdsContent.java#L75): Define a constant instead of duplicating this literal "#TOP_RIGHT_BOXAD" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsOnSpecialOasis.java [hunk 1]
```diff
@@ -7,75 +7,60 @@
...
-
-    verifySlotHasCorrectAd(
-        ads,
-        AdsContent.BOTTOM_LB,
-        adUnit,
-        PROJECT43_TEST_LINE_ITEM_ID
-    );
+    ads.setPageType("file");
+
+    FILE_PAGE_EXPECTED_LINE_ITEMS.put(AdsContent.TOP_LB, PROJECT43_TEST_LINE_ITEM_ID);
```
Comment (fraszczakszymon) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsOnSpecialOasis.java]: Can we use this hashmap in test scope only? It seems not needed as static class property

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsOnSpecialOasis.java [hunk 2]
```diff
@@ -7,75 +7,60 @@
...
-    ads.verifyGptIframe(adUnit, slotName, "gpt");
-    ads.verifyGptAdInSlot(slotName, lineItem, "");
+  private void specialPageExpectedDataProvider() {
+    SPECIAL_PAGE_EXPECTED_LINE_ITEMS.put(AdsContent.TOP_LB, "271491732");
+    SPECIAL_PAGE_EXPECTED_AD_UNITS.put(AdsContent.TOP_LB, "wka.life/_project43//special");
+  }
+
+  private void testSpecialPage(AdsBaseObject ads) {
+    ads.setPageType("special");
+    specialPageExpectedDataProvider();
```
Comment (fraszczakszymon) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsOnSpecialOasis.java]: Can we use this hashmap in test scope only? It seems not needed as static class property

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsContent.java#L72](https://github.com/Wikia/selenium-tests/blob/821ade42f769ecff2eab1283f626076e5fc92d22/src/test/java/com/wikia/webdriver/common/contentpatterns/AdsContent.java#L72): Define a constant instead of duplicating this literal "#BOTTOM_LEADERBOARD" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [AdsContent.java#L74](https://github.com/Wikia/selenium-tests/blob/821ade42f769ecff2eab1283f626076e5fc92d22/src/test/java/com/wikia/webdriver/common/contentpatterns/AdsContent.java#L74): Define a constant instead of duplicating this literal "#TOP_LEADERBOARD" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [AdsContent.java#L75](https://github.com/Wikia/selenium-tests/blob/821ade42f769ecff2eab1283f626076e5fc92d22/src/test/java/com/wikia/webdriver/common/contentpatterns/AdsContent.java#L75): Define a constant instead of duplicating this literal "#TOP_RIGHT_BOXAD" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsContent.java#L72](https://github.com/Wikia/selenium-tests/blob/31b18c0e44feab3dce86353ae340541219b14c5c/src/test/java/com/wikia/webdriver/common/contentpatterns/AdsContent.java#L72): Define a constant instead of duplicating this literal "#BOTTOM_LEADERBOARD" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [AdsContent.java#L74](https://github.com/Wikia/selenium-tests/blob/31b18c0e44feab3dce86353ae340541219b14c5c/src/test/java/com/wikia/webdriver/common/contentpatterns/AdsContent.java#L74): Define a constant instead of duplicating this literal "#TOP_LEADERBOARD" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [AdsContent.java#L75](https://github.com/Wikia/selenium-tests/blob/31b18c0e44feab3dce86353ae340541219b14c5c/src/test/java/com/wikia/webdriver/common/contentpatterns/AdsContent.java#L75): Define a constant instead of duplicating this literal "#TOP_RIGHT_BOXAD" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**